mongodump/metadata_dump.go

```diff
@@ -19,9 +19,10 @@ import (
 
 // Metadata holds information about a collection's options and indexes.
 type Metadata struct {
-	Options bson.M   `json:"options,omitempty"`
-	Indexes []bson.D `json:"indexes"`
-	UUID    string   `json:"uuid,omitempty"`
+	Options        bson.M   `json:"options,omitempty"`
+	Indexes        []bson.D `json:"indexes"`
+	UUID           string   `json:"uuid,omitempty"`
+	CollectionName string   `json:"collectionName"`
```

Comment:
I think we might want this to be `json:"collectionName,omitempty"`? I missed this at first.

Reply:
Will it ever be empty? Even for collection names that are < 238 bytes long, don't we want to write the name to the meta object?

Reply:
(Github won't let me reply to your other comment on this @mattChiaravalloti)

We'd be duplicating the collection name if we do this for short names (once in the .metadata.json filename, once in the .metadata.json `collectionName` field). We definitely could as a redundancy measure, but I'm not sure how much value it adds beyond that. We can see what @tfogo thinks.

Reply:
Yea this is a bit of an ambiguous point in the Detailed Design for this part. One bullet point seems to suggest adding the collection name here unconditionally. The next bullet point implies that the collection name can be taken from the filename if one is not found in the metadata. I'll defer to Tim here like you suggested.

Reply:
I think we should always include it. It won't always be used by mongorestore, but I think it's valuable to always have the same format.